scripts/helper.py

```diff
@@ -1,20 +1,28 @@
 """
-Library of auxiliary methods for scripts
-
-env: ppa-ocr
+Library of general-purpose auxiliary methods for stand-alone scripts
 """
 
 import os
 import sys
 import gzip
 import bz2
-import ftfy
 
 
 _char_conversion_map = {"ſ": "s"}
 _char_translation_table = str.maketrans(_char_conversion_map)
 
 
+def clean_htid(htid):
+    """
+    Returns the "clean" version of a HathiTrust volume id
+    """
+    lib_id, vol_id = htid.split(".", 1)
+    vol_id = vol_id.replace(":", "+")
+    vol_id = vol_id.replace("/", "=")
+    vol_id = vol_id.replace(".", ",")
+    return f"{lib_id}.{vol_id}"
+
```

Comment:
Is this a standard conversion? I'm not as familiar with the HTRC featureset code & approaches; was glancing at that and see how this is similar (or same?). Would benefit from slightly more context in the method explaining what the clean version is used for.

Is it worth using `translate` here instead of `replace` like you did for the other character replacement?

Reply:
This is the conversion that HTRC has used for extracted features (see this [utility function](https://github.com/htrc/htrc-feature-reader/blob/master/htrc_features/utils.py#L9)); it's also documented [here](https://htrc.atlassian.net/wiki/spaces/COM/pages/43296152/Stubbytree+directory+structure).

I can add the text that explains the conversion. It really is just remapping a few characters.

I can make the change to `translate`, since it's easy if not strictly necessary. It's probably slightly faster, although this isn't a place where we need to be concerned about performance.

Reply:
I had the same thought about `translate` - not strictly necessary.

So the clean id is for use with stubby tree / filenames? Are there other use cases?

Linking to the documentation is probably enough.

Reply:
Just to leave a record, this is really just following the cleaning procedure of pairtree / more generally making sanitizing htids so that filenames play nice with the filesystem.